src/cloud_client/dropbox/client.rs

```diff
@@ -0,0 +1,185 @@
+use crate::cloud_client::dropbox::parameters::list_folder::ListFolderParametersBuilder;
+use crate::cloud_client::dropbox::parameters::upload::UploadParametersBuilder;
+use crate::cloud_client::dropbox::responses::list_folder::ListFolderResult;
+use crate::cloud_client::CloudClient;
+use crate::errors::AppError;
+use reqwest::blocking::{Body, Client, ClientBuilder};
+use reqwest::header::{HeaderMap, HeaderValue, AUTHORIZATION, CONTENT_TYPE};
+use std::fs::File;
+use std::io::Read;
+use std::io::Write;
+use std::path::PathBuf;
+use tracing::{debug, info, instrument};
+
+static DROPBOX_API_HEADER: &str = "Dropbox-API-Arg";
+
+#[derive(Debug)]
+pub struct DropboxClient {
+    client: Client,
+}
+
+impl DropboxClient {
+    pub fn build() -> Result<DropboxClient, AppError> {
+        let token =
+            std::env::var("DROPBOX_ACCESS_TOKEN").map_err(|_| AppError::AbsentAccessToken)?;
+        let token = format!("Bearer {}", token);
+
+        let mut headers = HeaderMap::new();
+        headers.insert(
+            AUTHORIZATION,
+            HeaderValue::from_str(token.as_str()).map_err(|_| {
+                AppError::PrepareClient("unable to prepare authorization header value".to_string())
+            })?,
+        );
+
+        let client = ClientBuilder::new()
+            .default_headers(headers)
+            .build()
+            .map_err(|_| AppError::PrepareClient("unable to build request client".to_string()))?;
+
+        Ok(Self { client })
+    }
+}
+
+impl CloudClient for DropboxClient {
+    #[instrument(name = "Dropbox download", skip(self))]
+    fn download(&self, from_path: PathBuf, to_path: PathBuf) -> Result<(), AppError> {
+        info!("Downloading...");
+
+        let parameters = DownloadParametersBuilder::default()
+            .path(from_path)
+            .build()
+            .map_err(|_| AppError::PrepareRequestParameters)?;
+
+        let response = self
+            .client
+            .post(ApiUrl::Download.as_url())
+            .header(
+                DROPBOX_API_HEADER,
+                serde_json::to_string(&parameters).map_err(|_| AppError::PrepareRequest)?,
+            )
+            .send()
+            .map_err(|error| AppError::SendRequest(error.to_string()))?;
+
+        debug!("Response: {:?}", response);
+        match response.status().as_u16() {
+            200 => {
+                info!("File has been downloaded");
+
+                let bytes = response
+                    .bytes()
+                    .map_err(|_| AppError::Response("unable to get response body".to_string()))?;
+
+                let mut file = File::create(to_path).map_err(AppError::Io)?;
+                file.write_all(bytes.as_ref()).map_err(AppError::Io)?;
+
+                info!("File has been saved");
+                Ok(())
+            }
+            400 => Err(AppError::Request("check your input paths".to_string())),
+            401 => Err(AppError::Request("check your access token".to_string())),
```

Comment:
I think it's better to use constants provided by the reqwest crate instead of numbers for response status.

Reply:
Good idea, thanks/